src/main/java/io/github/intoto/models/PredicateType.java

```diff
@@ -0,0 +1,26 @@
+package io.github.intoto.models;
+
+import com.fasterxml.jackson.annotation.JsonValue;
+
+/**
+ * This enumeration is meant to represent the supported predicate types.
+ *
+ * @see <a
+ *     href="https://github.com/in-toto/attestation/blob/main/spec/README.md#predicate">in-toto/attestation/blob/main/spec/README.md#predicate</a>
+ */
+public enum PredicateType {
```

Comment:
optional: Make this an inner enum of the base Predicate class?

Reply:
The ipso facto standard is to not nest enums inside classes.

Reply:
Gotcha, fine to leave as is then.